### PR TITLE
Concept of "expected" errors and throttled loading errors

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: src/error.js [hunk 1]
```diff
@@ -195,13 +207,17 @@ function reportErrorToServer(message, filename, line, col, error) {
...
     hasNonAmpJs) {
+  let expected = false;
   if (error) {
     if (error.message) {
       message = error.message;
     } else {
       // This should never be a string, but sometimes it is.
       message = String(error);
     }
+    if (error.expected) {
```
Comment (cramforce) [inline src/error.js]: Document what this means and find a place to document how and when to use it.

Comment (dvoytenko) [inline src/error.js]: Added docs here and elsewhere inline. But do we have any overall documentation for errors in our repo?

Comment (erwinmombay) [inline src/error.js]: i dont think we have much documentation on errors, maybe we need one?

Comment (dvoytenko) [inline src/error.js]: I added docs in the `spec/amp-errors.md`.

File: spec/amp-errors.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+- `f=string` - the error's file name.
+- `l=number` - the error's line number.
+- `c=number` - the error's column number.
+- `r=string` - the document's referrer.
+- `ae=string` - accumulated/correlated error messages.
+- `fr=string` - the document's location fragment.
+
+## "Expected" errors
+
+The "expected" errors are marked with `&ex=1`.
```
Comment (cramforce) [inline spec/amp-errors.md]: Add "Expected errors are identified by the `.expected` property on the error object being true/"

File: spec/amp-errors.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# Error reporting in AMP
+
+AMP reports errors to the endpoint specified by the `errorReportingUrl` config
+property. E.g. `https://amp-error-reporting.appspot.com/r` for `cdn.ampproject.org`.
+
+The following fields are reported:
+
+- `v=string` - AMP version
+- `m=string` - the error message
+- `a=0|1` - whether the error is labeled as "user" or "dev" error.
```
Comment (cramforce) [inline spec/amp-errors.md]: Really only whether it was labeled as a user error. Maybe add a call example.

File: spec/amp-errors.md (hunk 2 above)
Comment (dvoytenko) [inline spec/amp-errors.md]: Done

File: spec/amp-errors.md (hunk 3 above)
Comment (dvoytenko) [inline spec/amp-errors.md]: Done

**Pull request merged**